addons/promote/jaxrs/src/main/java/org/commonjava/indy/promote/bind/jaxrs/PromoteAdminResource.java

```diff
@@ -0,0 +1,287 @@
+    {
+        return checkEnabledAnd( () -> {
+            try
+            {
+                validationsManager.parseRules();
+                return Response.ok().build();
+            }
+            catch ( PromotionException e )
+            {
+                logger.error( e.getMessage(), e );
+                return responseHelper.formatResponse( e );
+            }
+        } );
+    }
+
+    @ApiOperation( "Reload all rule-sets for promotion validation" )
+    @ApiResponse( code = 200, message = "", response = Response.class )
+    @Path( "/validation/reload/rulesets" )
+    @PUT
+    @Consumes( ApplicationContent.application_json )
+    public Response reloadRuleSets( final @Context HttpServletRequest servletRequest,
+                                    final @Context SecurityContext securityContext, final @Context UriInfo uriInfo )
+    {
+        return checkEnabledAnd( () -> {
+            try
+            {
+                validationsManager.parseRuleSets();
+                return Response.ok().build();
+            }
+            catch ( PromotionException e )
+            {
+                logger.error( e.getMessage(), e );
+                return responseHelper.formatResponse( e );
+            }
+        } );
+    }
+
+    @ApiOperation( "Reload all rules & rule-sets for promotion validation" )
+    @ApiResponse( code = 200, message = "", response = Response.class )
+    @Path( "/validation/reload/all" )
+    @PUT
+    @Consumes( ApplicationContent.application_json )
+    public Response reloadAll( final @Context HttpServletRequest servletRequest,
+                               final @Context SecurityContext securityContext, final @Context UriInfo uriInfo )
+    {
+        return checkEnabledAnd( () -> {
+            try
+            {
+                validationsManager.parseRuleBundles();
+                return Response.ok().build();
+            }
+            catch ( PromotionException e )
+            {
+                logger.error( e.getMessage(), e );
+                return responseHelper.formatResponse( e );
+            }
+        } );
+    }
+
+    @ApiOperation( "Get all rules' names" )
+    @ApiResponses( { @ApiResponse( code = 200, response = Response.class,
+                                   message = "All promotion validation rules' definition" ),
+                           @ApiResponse( code = 404, message = "No rules are defined" ) } )
+    @Path( "/validation/rules/all" )
+    @GET
+    @Consumes( ApplicationContent.application_json )
+    public Response getAllRules( final @PathParam( "name" ) String ruleName,
+                                 final @Context HttpServletRequest servletRequest,
+                                 final @Context SecurityContext securityContext, final @Context UriInfo uriInfo )
+    {
+        return checkEnabledAnd( () -> {
+            Map<String, ValidationRuleDTO> rules = validationsManager.toDTO().getRules();
+            if ( rules.isEmpty() )
```

Comment:
Looks this should be `!rules.isEmpty()`.

Reply:
Yup, it is a typo. Will be fixed in new commit.